src/pages/login-helper.js

```diff
@@ -1,49 +1,78 @@
-import { jwtDecode } from 'jwt-decode';
+// login-helper.js
+const jwtDecode = require('jwt-decode');
+
+let decoded;
 
 function authenticate(jwt, cb) {
-    if (typeof window !== "undefined") {
-        sessionStorage.setItem('jwt', jwt);
+  if (typeof window !== "undefined") {
+    const existingJwt = sessionStorage.getItem('jwt');
+    const existingUsers = JSON.parse(sessionStorage.getItem('users')) || [];
 
-        let decoded = jwtDecode(jwt);
-        sessionStorage.setItem('username', decoded.username);
+    if (!existingJwt && existingUsers.length === 0) {
+      // First user registration, set as admin
+      sessionStorage.setItem('getIsAdmin', 'true');
+    } else {
+      decoded = jwtDecode(jwt);
+      sessionStorage.setItem('getIsAdmin', decoded.isAdmin);
     }
-    cb();
+
+    sessionStorage.setItem('jwt', jwt);
+    sessionStorage.setItem('username', decoded.username);
+  }
+  cb();
 }
 
 function checkToken(token) {
-    let decoded = jwtDecode(token);
-    if (decoded.exp < Date.now() / 1000) {
-        sessionStorage.removeItem('jwt');
-        sessionStorage.removeItem('username');
-        return false;
-    }
-    return true;
+  let decoded = jwtDecode(token);
+  if (decoded.exp < Date.now() / 1000) {
+    sessionStorage.removeItem('jwt');
+    sessionStorage.removeItem('username');
+    sessionStorage.removeItem('getIsAdmin');
+    return false;
+  }
+  return true;
 }
 
 function isAuthenticated() {
-    if (typeof window == "undefined") {
-        return false
-    }
-    let token = sessionStorage.getItem('jwt');
+  if (typeof window === "undefined") {
+    return false;
+  }
+  let token = sessionStorage.getItem('jwt');
 
-    if (token) {
-        return checkToken(token);
-    } else {
-        return false;
-    }
+  if (token) {
+    return checkToken(token);
+  } else {
+    return false;
+  }
 }
 
 function logout() {
-    if (typeof window !== "undefined") {
-        sessionStorage.removeItem('jwt');
-        sessionStorage.removeItem('username');
-    }
+  if (typeof window !== "undefined") {
+    sessionStorage.removeItem('jwt');
+    sessionStorage.removeItem('username');
+    sessionStorage.removeItem('getIsAdmin');
+  }
 }
 
 function getUsername() {
-    if (typeof window !== "undefined") {
-        return sessionStorage.getItem('username');
-    }
+  if (typeof window !== "undefined") {
+    return sessionStorage.getItem('username');
+  }
+}
+
+function getIsAdmin() {
+  if (typeof window !== 'undefined') {
+    return sessionStorage.getItem('getIsAdmin') === 'true';
+  }
+}
+
+async function promoteToAdmin(username) {
+  try {
+    //  promote the user to admin
```

Comment:
Then this should make a request to the backend to make the user an admin.
Also sus chatGPT usage